modules/ml/src/rtrees.cpp

```diff
@@ -235,35 +236,31 @@ class DTreesImplForRTrees : public DTreesImpl
                     oobperm.resize(n_oob);
                     for( i = 0; i < n_oob; i++ )
                         oobperm[i] = oobidx[i];
+                    std::random_shuffle(oobperm.begin(), oobperm.end()); //Randomly shuffle indices so we can permute features
```

Comment:
It is hard to test algorithms with `std::random_shuffle` (results in tests should be reproducible, OpenCV "rng" object can guarantee this).
Implementation with reused "rng":
```
for (i = n_oob - 1; i > 0; --i)  // simulate std::random_shuffle
{
    int j = rng.uniform(0, i + 1);
    std::swap(oobperm[i], oobperm[j]);
}
```